pkg/executables/kubectl.go

```diff
@@ -483,6 +496,31 @@ func (k *Kubectl) GetMachines(ctx context.Context, cluster *types.Cluster, clust
 	return response.Items, nil
 }
 
+type machineSetResponse struct {
+	Items []clusterv1.MachineSet `json:"items,omitempty"`
+}
+
+func (k *Kubectl) GetMachineSets(ctx context.Context, machineDeploymentName string, cluster *types.Cluster) ([]clusterv1.MachineSet, error) {
```

Comment:
nit: `GetMachineSetsForMachineDeployment` or `GetMachineSetsByMachineDeploymentName`

Reply:
will change this when I add the unit tests